crispr_sites/crispr_sites.cpp

```diff
@@ -349,20 +354,29 @@ void scan_stdin() {
         }
         last = *it;
     }
-
+    
     cerr << "Outputting " << guides << " unique guides." << endl;
     last = 0;
     char obuf[k-1];
     obuf[k-2] = 0;
     obuf[k-3] = '\n';
+
+    if (output_counts) {
+        obuf[k-3] = '\t';
+    }
+
+    int32_t current_count = 0;
     for (auto it = results.begin();  it != results.end();  ++it) {
-        if (*it != last) {
+        ++current_count;
+        if (*next(it) != *it) {
```

Comment:
When `it` points to the last element of the array, `*next(it)` is dereferencing a pointer to uninitialized memory.   Often the value there will be 0 which is guaranteed to be different from the value of the last array element, and that makes your code work.  That's why you haven't caught this bug in your testing.    But, if you run on a different operating system (particularly Windows) you may notice memory is not all zeroes by default, and this code could miss the last count.   And, if the vector had more complex elements, this could crash.

Fortunately you can fix the problem with one very small change.   Make the if guard here like so: 
```
    if (next(it) == results.end() || *next(it) != *it) {
        ...
    }
```

Reply:
Ah! I didn't know that dereferencing an iterator pointing to results.end() was undefined Fixed now!